Automattic/jetpack
# cli: Fix ENOENT in `jetpack rsync`

#### Changes proposed in this Pull Request:

If a broken symlink somehow winds up existing in a plugin's `vendor/` or `jetpack_vendor/`, composer won't clean it up. Then if you try to `jetpack rsync` the affected plugin, it'll wind up failing with a cryptic "spawn git ENOENT" when it tries to run git in the nonexistent target directory.

We can avoid that by checking for broken symlinks before trying to follow them in the `jetpack rsync` logic. Rsync itself will produce a more helpful message.

#### Other information:


#### Jetpack product discussion


p1672952547856599-slack-CDLH4C1UZ

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create a broken symlink in a `jetpack_vendor` dir, e.g with `cd projects/plugins/jetpack/jetpack_vendor/automattic && ln -s ../../../../packages/promote-posts/`.
* Run `jetpack rsync jetpack`. The rsync should run to completion, printing a "symlink has no referent" message pointing to the problematic symlink early on.
